src/gleam/list.gleam

```diff
@@ -817,7 +829,19 @@ pub fn all(in list: List(a), satisfying predicate: fn(a) -> Bool) -> Bool {
 pub fn any(in list: List(a), satisfying predicate: fn(a) -> Bool) -> Bool {
   case list {
     [] -> False
-    [x, ..rest] -> predicate(x) || any(rest, predicate)
+    list -> any_tail_recursive(list, predicate, False)
+  }
+}
+
+fn any_tail_recursive(
+  list: List(a),
+  predicate: fn(a) -> Bool,
+  accumulator: Bool,
+) -> Bool {
+  case list {
+    [] -> accumulator
+    [x, ..rest] ->
+      any_tail_recursive(rest, predicate, accumulator || predicate(x))
```

Comment:
Total Gleam noob here, just reading PR's and issues to learn, so forgive me if this is a dumb question. Won't this run on the entire list even if say the first element fits the predicate? Should this short circuit if the predicate is `True` ? Or will that break the TCO?

Reply:
I think(?!) you are right, and I think(?!) I have fixed it in the commit I just pushed: https://github.com/gleam-lang/stdlib/pull/322/commits/cb4acbcfae42ee02ec36dbe7ae286f1307491021